test/cctest/test_node_debug_support.cc

```diff
@@ -0,0 +1,127 @@
+/* {{debug_symbols_file}} */
+
+/*
+ * This file is a template used by ./tools/gen-postmortem-metadata.py to
+ * generate a proper test file
+ */
+
+#include "gtest/gtest.h"
+#include "node_test_fixture.h"
+#include "node.h"
+#include "node_internals.h"
+#include "udp_wrap.h"
+#include "v8.h"
+#include "req_wrap-inl.h"
+#include "tracing/agent.h"
+
+const int node::Environment::kContextEmbedderDataIndex;
+
+
+class DebugSymbolsTest : public EnvironmentTestFixture {
+};
+
+
+class TestHandleWrap : public node::HandleWrap {
+  public:
+   size_t self_size() const override { return sizeof(*this); }
+
+   TestHandleWrap(node::Environment* env, v8::Local<v8::Object> object, uv_handle_t* handle) :
+     node::HandleWrap(env, object, handle, node::AsyncWrap::PROVIDER_TIMERWRAP) {}
+};
+
+
+class TestReqWrap : public node::ReqWrap<uv_req_t> {
+  public:
+   size_t self_size() const override { return sizeof(*this); }
+
+   TestReqWrap(node::Environment* env, v8::Local<v8::Object> object) :
+     node::ReqWrap<uv_req_t>(env, object, node::AsyncWrap::PROVIDER_TIMERWRAP) {}
+};
+
+TEST_F(DebugSymbolsTest, ContextEmbedderDataIndex) {
+  EXPECT_EQ(nodedbg_environment_context_idx_embedder_data, node::Environment::kContextEmbedderDataIndex);
+}
+
+TEST_F(DebugSymbolsTest, BaseObjectPersistentHandle) {
+  const v8::HandleScope handle_scope(isolate_);
+  const Argv argv;
+  Env env {handle_scope, isolate_, argv, this};
+
+  v8::Local<v8::Object> object = v8::Object::New(isolate_);
+  node::BaseObject *obj = new node::BaseObject(*env, object);
```

Comment:
Leaks memory.  Use `std::unique_ptr`.

Reply:
I tried using `std::unique_ptr` and `delete`, but both are throwing an error at the end of the test because the referenced `v8::Object` still exists:

```
[----------] 6 tests from DebugSymbolsTest
[ RUN      ] DebugSymbolsTest.ContextEmbedderDataIndex
[       OK ] DebugSymbolsTest.ContextEmbedderDataIndex (4 ms)
[ RUN      ] DebugSymbolsTest.BaseObjectPersistentHandle
[18333]: ../src/base_object-inl.h:46:virtual node::BaseObject::~BaseObject(): Assertion `persistent_handle_.IsEmpty()' failed.
 1: node::Abort() [./out/Release/cctest]
 2: 0x14ef3ef [./out/Release/cctest]
 3: DebugSymbolsTest_BaseObjectPersistentHandle_Test::TestBody() [./out/Release/cctest]
 4: 0x128dabc [./out/Release/cctest]
 5: 0x128dd3d [./out/Release/cctest]
 6: 0x128df15 [./out/Release/cctest]
 7: 0x128f563 [./out/Release/cctest]
 8: testing::UnitTest::Run() [./out/Release/cctest]
 9: main [./out/Release/cctest]
10: __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
11: _start [./out/Release/cctest]
Aborted (core dumped)
```

The failing check is:

```c++
inline BaseObject::~BaseObject() {
  CHECK(persistent_handle_.IsEmpty());
}
```

I'll try it again tomorrow, but suggestions are appreciated.

Reply:
Right, the destructor expects to be invoked by the garbage collector.  You can cheat and call `obj->persistent().Reset()` when you're done with it.